src/DataStreams.jl

```diff
@@ -18,346 +18,424 @@ A `Data.Schema` describes a tabular dataset (i.e. a set of optionally named, typ
 `Data.Schema` allow `Data.Source` and `Data.Sink` to talk to each other and prepare to provide/receive data through streaming.
 `Data.Schema` fields include:
 
- * A boolean type parameter that indicates whether the # of rows is known in the `Data.Source`; this is useful as a type parameter to allow `Data.Sink` and `Data.streamto!` methods to dispatch. Note that the sentinel value `-1` is used as the # of rows when the # of rows is unknown.
  * `Data.header(schema)` to return the header/column names in a `Data.Schema`
  * `Data.types(schema)` to return the column types in a `Data.Schema`; `Nullable{T}` indicates columns that may contain missing data (null values)
- * `Data.size(schema)` to return the (# of rows, # of columns) in a `Data.Schema`
-
-`Data.Source` and `Data.Sink` interfaces both require that `Data.schema(source_or_sink)` be defined to ensure
-that other `Data.Source`/`Data.Sink` can work appropriately.
+ * `Data.size(schema)` to return the (# of rows, # of columns) in a `Data.Schema`; note that # of rows may be `null`, meaning unknown
 """
-type Schema{RowsAreKnown}
+mutable struct Schema{R, T}
     header::Vector{String}       # column names
-    types::Vector{DataType}      # Julia types of columns
-    rows::Int                    # number of rows in the dataset
+    # types::T                   # Julia types of columns
+    rows::(?Int)                      # number of rows in the dataset
```

Comment:
Maybe this is the kind of situation where we'd better keep using `Nullable` so that you're forced to handle the null case explicitly?